### DISCUSSION TITLE
New devices not detected

### DISCUSSION THREAD

**Discussion by bouillou7** [milo526/homebridge-tuya-web#487] [Q&A]
Hello, I added new devices in tuya and homebridge does not detect them. This is a plug and LED strip
How to do?

**Comment by milo526** [comment]
Please post the result of [the steps under "How to check whether the api this library uses can control your device".](https://github.com/milo526/homebridge-tuya-web#how-to-check-whether-the-api-this-library-uses-can-control-your-device)

Without this it is hard for me to help you with regards to this specific issue.

**Reply by bouillou7** [reply]
Getting credentials
Got credentials
{   'access_token': 'EUheu15735116176426pf7nFNJlELuw0b',
    'expires_in': 864000,
    'refresh_token': 'EUheu15735116176426pf7nEya8bBuwlP',
    'token_type': 'bearer'}
Getting devices
Got devices
{   'header': {'code': 'SUCCESS', 'payloadVersion': 1},
    'payload': {   'devices': [   {   'data': {   'brightness': '110',
                                                  'color_mode': 'white',
                                                  'color_temp': 19247,
                                                  'online': True,
                                                  'state': 'false'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/15984362377f0fef3c162.png',
                                      'id': 'bf1e6e8fb197c6b3fdbpoa',
                                      'name': 'Chevet Valérie '},
                                  {   'data': {   'brightness': '100',
                                                  'color_mode': 'white',
                                                  'color_temp': 36294,
                                                  'online': False,
                                                  'state': 'true'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/16381491795f6f8d05099.png',
                                      'id': 'bf3e6126d5a17aef92ofe1',
                                      'name': 'Plafond Bureau'},
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1518345874298thrlS/1576123591fa6c6c029d7.png',
                                      'id': '534346768caab5e7be0f',
                                      'name': 'Chauffage Salle de bain '},
                                  {   'data': {   'brightness': '641',
                                                  'color_mode': 'white',
                                                  'color_temp': 26588,
                                                  'online': False,
                                                  'state': 'true'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/161896948897ac1b70be2.png',
                                      'id': 'bf55bf2eed31d2c0300ska',
                                      'name': 'Plafond Toilettes '},
                                  {   'data': {   'brightness': '300',
                                                  'color_mode': 'white',
                                                  'online': False,
                                                  'state': 'true'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/bay16279833154642Sqp/1643375661381286315e8.png',
                                      'id': 'bf17a76368da7f7c1c4dx6',
                                      'name': 'Spot Piscine'},
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/162147598905ecd9cb320.png',
                                      'id': 'bfbab4379664885e8cu87c',
                                      'name': 'Chauffage Romän'},
                                  {   'data': {   'online': False,
                                                  'state': 'false'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/1634819612bbe427a304e.png',
                                      'id': '88083846c44f33e40726',
                                      'name': 'Guirlande Noël'},
                                  {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/162147598905ecd9cb320.png',
                                      'id': 'bf3a1f9cc872e19c99n7eu',
                                      'name': 'TV Sacha '},
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/1604989149786bb63bb14.png',
                                      'id': 'bfc2f56d545fcdfc0bcde0',
                                      'name': 'Chauffage Salon'},
                                  {   'data': {   'brightness': '930',
                                                  'color_mode': 'colour',
                                                  'color_temp': 20412,
                                                  'online': True,
                                                  'state': 'false'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/15984362377f0fef3c162.png',
                                      'id': 'bf16910e961a7b0744rh0s',
                                      'name': 'Chevet Salon'},
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/1604989149786bb63bb14.png',
                                      'id': 'bf6b9b8744aead4599qzqk',
                                      'name': 'Arbre Salle'},
                                  {   'data': {   'brightness': '1000',
                                                  'color_mode': 'white',
                                                  'color_temp': 36294,
                                                  'online': True,
                                                  'state': 'false'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/1595465809b265540ffda.png',
                                      'id': 'bf8b519ce966399c4fezyj',
                                      'name': 'Spot Garage'},
                                  {   'data': {'online': False, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1518345874298thrlS/1576123591fa6c6c029d7.png',
                                      'id': '534346768caab5e868b5',
                                      'name': 'Sèche-linge '},
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1506337792739cXrIR/1582528372f1972024f70.jpg',
                                      'id': '54625385d8bfc0f89b60',
                                      'name': 'Chauffage Salle à Manger'},
                                  {   'data': {   'brightness': '10',
                                                  'color_mode': 'colour',
                                                  'color_temp': 15188,
                                                  'online': True,
                                                  'state': 'false'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/15984362377f0fef3c162.png',
                                      'id': 'bfbfeb5c4a3d88a5aai4ar',
                                      'name': 'Chevet Johnny'},
                                  {   'data': {   'brightness': '1000',
                                                  'color_mode': 'colour',
                                                  'color_temp': 1000,
                                                  'online': True,
                                                  'state': 'false'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1528449959696EWXeY/160334777173054206318.jpg',
                                      'id': 'bfe1d0c46d43c51d7c8fke',
                                      'name': 'Spot Salon'},
                                  {   'data': {   'brightness': '1000',
                                                  'color_mode': 'colour',
                                                  'color_temp': 18647,
                                                  'online': True,
                                                  'state': 'false'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1528449959696EWXeY/160334777173054206318.jpg',
                                      'id': 'bffbf2ba32b763708epsi8',
                                      'name': 'Spot Salle à manger'},
                                  {   'data': {   'brightness': '330',
                                                  'color_mode': 'colour',
                                                  'color_temp': 12400,
                                                  'online': True,
                                                  'state': 'true'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/160914662513804f14116.png',
                                      'id': 'bff6a942149366e13esapf',
                                      'name': 'Meuble Cuisine'},
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/15269692784aqaowp96ea28vdjkdw0qw7b9_0.png',
                                      'id': '84004508807d3a478448_1',
                                      'name': 'Lumière Nuit'},
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/15269692784aqaowp96ea28vdjkdw0qw7b9_0.png',
                                      'id': '84004508807d3a478448_2',
                                      'name': 'Chauffage Sacha'},
                                  {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/15269692784aqaowp96ea28vdjkdw0qw7b9_0.png',
                                      'id': '84004508807d3a478448_3',
                                      'name': 'Multiprise Téléphone'},
                                  {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/15269692784aqaowp96ea28vdjkdw0qw7b9_0.png',
                                      'id': '84004508807d3a478448_7',
                                      'name': 'Sacha USB'},
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/15269692784aqaowp96ea28vdjkdw0qw7b9_0.png',
                                      'id': '85502354807d3a50bfb8_1',
                                      'name': 'Lumière Tortue'},
                                  {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/15269692784aqaowp96ea28vdjkdw0qw7b9_0.png',
                                      'id': '85502354807d3a50bfb8_2',
                                      'name': 'Chauffage Tortue'},
                                  {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/15269692784aqaowp96ea28vdjkdw0qw7b9_0.png',
                                      'id': '85502354807d3a50bfb8_3',
                                      'name': 'Ampoule Tortue'},
                                  {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/15269692784aqaowp96ea28vdjkdw0qw7b9_0.png',
                                      'id': '85502354807d3a50bfb8_7',
                                      'name': 'Tortue USB'},
                                  {   'data': {'online': False, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/16195941640c3fb4ffbca.jpg',
                                      'id': '15727703c4dd57115a2e_1',
                                      'name': 'Switch 1'},
                                  {   'data': {'online': False, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/16195941640c3fb4ffbca.jpg',
                                      'id': '15727703c4dd57115a2e_2',
                                      'name': 'PAC Piscine'},
                                  {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/15269692784aqaowp96ea28vdjkdw0qw7b9_0.png',
                                      'id': '76565238807d3a1c722b_1',
                                      'name': 'Imprimante Romän'},
                                  {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/15269692784aqaowp96ea28vdjkdw0qw7b9_0.png',
                                      'id': '76565238807d3a1c722b_2',
                                      'name': 'Rien Romän'},
                                  {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/15269692784aqaowp96ea28vdjkdw0qw7b9_0.png',
                                      'id': '76565238807d3a1c722b_3',
                                      'name': 'Rien Romän 2'},
                                  {   'data': {'online': True, 'state': True},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/15269692784aqaowp96ea28vdjkdw0qw7b9_0.png',
                                      'id': '76565238807d3a1c722b_7',
                                      'name': 'Romän USB'},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'Dj2Q5Q8BiHbiidBm',
                                      'name': 'Allumage Chauffage Séjour 21'},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'zcTkEeegALZw5d1M',
                                      'name': 'Coupure chauffage séjour 22'},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'UDNQJ1nhGRy4Gnd0',
                                      'name': 'Chauffage Parents '},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'vx7oXdpAXN47VlcI',
                                      'name': 'Chauffage Romän'},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'PoWoOi6lRbHtMr2K',
                                      'name': 'Chauffage Sacha'},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'ykZZ3GZ9M3Z5nKb6',
                                      'name': 'Chauffage Salle de bain '},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'Al9AUcA99XXAcOvR',
                                      'name': 'Extinction lumière piscine '
                                              'Nuit'},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': '0u9OpA4x7br1qQQy',
                                      'name': 'Allumage lumière piscine Nuit'},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'X8vLIz8zLbfm93Vd',
                                      'name': 'Extinction Chauffage Laverie '},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'A3WJzJgzjmbsoNMP',
                                      'name': 'Allumage Chauffage Laverie '},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'KfErn31l20FKGRSp',
                                      'name': 'Niveau de batterie '
                                              'Thermomètre '},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'EyA24YP5IrWguTXK',
                                      'name': 'Éclairage Meuble cuisine au '
                                              'Coucher du soleil '},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'BVzl5ka0XHXojMEr',
                                      'name': 'Extinction Meuble Cuisine au le '
                                              'lever du soleil '},
                                  {   'data': {},
                                      'dev_type': 'scene',
                                      'ha_type': 'scene',
                                      'id': 'rxwDY1rxu3foFnAj',
                                      'name': 'Extinction lumière Sacha '}],
                   'scenes': []}}
!!! NOW REMOVE THIS FILE, SO YOUR CREDENTIALS (username, password) WON'T LEAK !!!


**Comment by milo526** [comment]
And do you see the name of the newly installed ledstrip in the above list?

If not; it is not supported by the API that this plugin uses.

**Reply by bouillou7** [reply]
No, but it's the same model that appears in the list. Same for the switch.
For the ledstrip, it's the same of:
'id': 'bff6a942149366e13esapf',
                                      'name': 'Meuble Cuisine'},
                                  {   'data': {'online': True, 'state': False},
                                      'dev_type': 'switch',
                                      'ha_type': 'switch',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/15269692784aqaowp96ea28vdjkdw0qw7b9_0.png',
For the switch, the same of:
'id': 'bfc2f56d545fcdfc0bcde0',
                                      'name': 'Chauffage Salon'},
                                  {   'data': {   'brightness': '930',
                                                  'color_mode': 'colour',
                                                  'color_temp': 20412,
                                                  'online': True,
                                                  'state': 'false'},
                                      'dev_type': 'light',
                                      'ha_type': 'light',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay1541056239985fDGjj/15984362377f0fef3c162.png',
so i don't understand

**Reply by milo526** [reply]
This plugin shows all the devices that are listed. If the device is not listed, this plugin does not receive any information about it and therefore cannot show it.
I don't know why some of these devices show and some do not. Did you double-check that they are added to the correct tuya account? Do you see the devices in the Tuya app?
It might be some arbitrary limit on the number of devices they show in this API but I have not heard of anything like that from Tuya before.
